src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -68,22 +70,39 @@
     private final Instant from;
     private Instant to;
 
+    private CrawlerTime crawlerTime;
+
     private final Duration defaultIntervalLength;
 
     private long numberOfEvents;
     private long numberOfMessages;
 
     private final boolean floatingToTime;
+    private final boolean workAlone;
+    private boolean reachedTo;
 
     private long sleepTime;
 
+    private String crawlerType;
+    private int batchSize;
+    private DataServiceInfo info;
+    private CrawlerId crawlerId;
+
     private static final String EVENTS = "EVENTS";
     private static final String MESSAGES = "MESSAGES";
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Crawler.class);
 
     private Interval interval;
 
+    public Crawler(@NotNull CradleStorage storage, @NotNull DataServiceService dataService,
+                   @NotNull DataProviderService dataProviderService, @NotNull CrawlerConfiguration configuration,
+                   CrawlerTime crawlerTime) {
+        this(storage, dataService, dataProviderService, configuration);
+
+        this.crawlerTime = crawlerTime;
```

Comment:
Please, check for `null`. Also, I think that the constructor with main initialization should have the CrawlerTime parameter.
So, you will have a main constructor
```java
Crawler(@NotNull CradleStorage storage, @NotNull DataServiceService dataService,
                   @NotNull DataProviderService dataProviderService, @NotNull CrawlerConfiguration configuration,
                   CrawlerTime crawlerTime) {
/// init
}
```

And the secondary
```java
Crawler(@NotNull CradleStorage storage, @NotNull DataServiceService dataService,
                   @NotNull DataProviderService dataProviderService, @NotNull CrawlerConfiguration configuration) {
                   this(storage, dataService, dataProviderService, configuration, new CrawlerTimeImpl());
}
```